src/utils/fileUtils.ts

```diff
@@ -0,0 +1,64 @@
+import * as path from "path";
+import { TextDocument, window, workspace, WorkspaceFolder } from "vscode";
+
+/**
+ * Return the current workspace uri from root
+ *
+ * @param currentFile the opened TextDocument
+ * @returns the path from root of the current workspace
+ */
+export function getWorkspaceUri(currentFile: TextDocument): string {
+  const currentWorkspace: WorkspaceFolder = (currentFile && currentFile.uri) ?  workspace.getWorkspaceFolder(currentFile.uri) : undefined;
+ return ((currentWorkspace && currentWorkspace.uri.fsPath) || (workspace.workspaceFolders && workspace.workspaceFolders[0].uri.fsPath));
+}
+
+/**
+ * Return the current workspace absolute uri
+ *
+ * @param currentFile the opened TextDocument
+ * @returns the absolute path of the current workspace
+ */
+export function getAbsoluteWorkspaceUri(currentFile: TextDocument): string {
+  const currentWorkspace: WorkspaceFolder = (currentFile && currentFile.uri) ?  workspace.getWorkspaceFolder(currentFile.uri) : undefined;
+  return ((currentWorkspace && currentWorkspace.uri) || (workspace.workspaceFolders && workspace.workspaceFolders[0].uri)).toString();
+}
+
+/**
+ * Retrieve the text document that is open in the editor
+ *
+ * @returns the text document open in the editor
+ */
+export function getCurrentFile(): TextDocument {
+  return (window.activeTextEditor && window.activeTextEditor.document && window.activeTextEditor.document.languageId === 'xml') ?  window.activeTextEditor.document : undefined;
+}
+
+/**
+ * Return the uri of the current file
+ *
+ * @param currentFile the opened TextDocument
+ * @returns the uri of the current file
+ */
+export function getCurrentFileUri(currentFile: TextDocument): string {
```

Comment:
Rename with getFileUri

Reply:
This contradicts https://github.com/redhat-developer/vscode-xml/pull/584#discussion_r690658037, is that alright?

Reply:
@angelozerr

Reply:
Yes sorry rename with getFilePath